openassessment/xblock/test/test_grade.py

```diff
@@ -320,28 +320,38 @@ def test_grade_waiting(self, xblock, data):
         # Verify that we're on the waiting template
         self.assertIn(data["expected_response"], resp.decode('utf-8').lower())
 
+    @ddt.data('self', 'peer', 'both')
     @scenario('data/grade_incomplete_scenario.xml', user_id='Bunk')
-    def test_grade_incomplete_missing_self(self, xblock):
-        # Graded peers, but haven't completed self assessment
-        self.create_submission_and_assessments(
-            xblock, self.SUBMISSION, [self.PEERS[0]], [PEER_ASSESSMENTS[0]], None
-        )
-        resp = self.request(xblock, 'render_grade', json.dumps(dict()))
-
-        # Verify that we're on the right template
-        self.assertIn(u'not completed', resp.decode('utf-8').lower())
+    def test_grade_incomplete_missing(self, xblock, missing_step):
```

Comment:
This method was already doing all the setup for "incomplete" scenarios, so I elected to just add the new message tests here instead of duplicating all that setup in `test_submission.py`. Any strong objections?